core/rio/api/src/main/java/org/eclipse/rdf4j/rio/Rio.java

```diff
@@ -323,6 +372,30 @@ public static Model parse(Reader reader, RDFFormat dataFormat, Resource... conte
 				new ParseErrorLogger(), contexts);
 	}
 
+	/**
+	 * Adds RDF data from a {@link Reader} to a {@link Model}, optionally to one or more named contexts. <b>Note: using
+	 * a Reader to upload byte-based data means that you have to be careful not to destroy the data's character encoding
+	 * by enforcing a default character encoding upon the bytes. If possible, adding such data using an InputStream is
+	 * to be preferred.</b>
+	 *
+	 * @param reader     A Reader from which RDF data can be read.
+	 * @param dataFormat The serialization format of the data.
+	 * @param settings   The {@link ParserConfig} containing settings for configuring the parser.
+	 * @param contexts   The contexts to add the data to. If one or more contexts are specified the data is added to
+	 *                   these contexts, ignoring any context information in the data itself.
+	 * @return A {@link Model} containing the parsed statements.
+	 * @throws IOException                  If an I/O error occurred while reading from the reader.
+	 * @throws UnsupportedRDFormatException If no {@link RDFParser} is available for the specified RDF format.
+	 * @throws RDFParseException            If an error was found while parsing the RDF data.
+	 * 
+	 * @since 4.0.0
+	 */
+	public static Model parse(Reader reader, RDFFormat dataFormat, ParserConfig settings, Resource... contexts)
```

Comment:
Seeing these changes I remember that I wanted us to have dedicated methods for StringReader and StringWriter  that don't throw IOException. I noticed that Jena has this and it makes the API just that little bit simpler and cleaner to use.

Reply:
Nice idea!